src/rng.cpp

```diff
@@ -1,19 +1,30 @@
-#include "output.h"
 #include "rng.h"
-#include <stdlib.h>
+#include <random>
 
-long rng(long val1, long val2)
+namespace {
+std::mt19937 random_generator {std::random_device{}()};
```

Comment:
Sharable seeds for challenges / tournaments? Game replays?
There are no stand-out features that immediately come to mind, but it's something to put on the back burner and maybe consider.